docs/contributing.md

```diff
@@ -197,23 +222,21 @@ row.event_timestamp.CopyFrom(timestamp)
 # The format is [FEATURE_NAME]:[VERSION]
 row.feature_set = "driver:1"
 
-producer.send("feast-driver-features", row.SerializeToString())
+producer.send("your-kafka-topic", row.SerializeToString())
 producer.flush()
 logger.info(row)
 EOF
 
 # Check that the ingested feature rows can be retrieved from Feast serving
 grpc_cli call localhost:6566 GetOnlineFeatures '
-feature_sets {
+features {
+  project: "your_project_name"
   name: "driver"
   version: 1
 }
-entity_dataset {
-  entity_names: "driver_id"
-  entity_dataset_rows {
-    entity_ids {
-      int64_val: 1234
-    }
+entity_rows {
```

Comment:
```
fields {
  "driver_id": 1234
}
```
doesn't work,  
```
[libprotobuf ERROR google/protobuf/text_format.cc:317] Error parsing text-format feast.serving.GetOnlineFeaturesRequest: 10:11: Message type "feast.serving.GetOnlineFeaturesRequest.EntityRow.FieldsEntry" has no field named "name".
Failed to convert text format to proto.
```

Could you help check this part? @woop 
https://github.com/gojek/feast/blob/bb960a7f540dac430ca7bce972b33649ff6e0dc2/protos/feast/serving/ServingService.proto#L102

Reply:
Hi @Jeffwan, the message structure has changed since then, it should be something like this
```
grpc_cli call localhost:6566 GetOnlineFeatures '
entity_rows {
  fields {
    key: "driver_id"
    value {
      int64_val: 1234
    }
  }
}
features {
  project: "your_project_name"
  name: "city"
  version: 1
  max_age {
    seconds: 3600
  }
}
'
```

Reply:
Yea I think this is the only change required. Otherwise it looks good :)

Reply:
Great! Thanks @david30907d and @woop. I update it and it works for me now.

Reply:
One more question, when I try to get feature. Seems response is empty. Should it be `your_project_name/city:1` or `your_project_name/driver:1`. Another thing I notice is redis is empty and I didn't find any consumer for kafka cluster.  I am very new to feast. Could I know if we use `DirectRunner` locally,  feast-core will kick off Beam Job as a thread to consume the data? 

```
    key: "your_project_name/city:1"
    value {
    }
```

full logs
```
grpc_cli call localhost:6566 GetOnlineFeatures '
features {
  project: "your_project_name"
  name: "city"
  version: 1
  max_age {
    seconds: 360000
  }
}
entity_rows {
  fields {
    key: "driver_id"
    value {
      int64_val: 1234
    }
  }
}
'
connecting to localhost:6566
field_values {
  fields {
    key: "driver_id"
    value {
      int64_val: 1234
    }
  }
  fields {
    key: "your_project_name/city:1"
    value {
    }
  }
}
```

Reply:
Thanks for the update @Jeffwan 

> Could I know if we use DirectRunner locally, feast-core will kick off Beam Job as a thread to consume the data?

Yes, it should start the job.

If there is no data in Redis, then probably its a consumption problem. Do you see any subprocesses for Beam running? Which feature sets have you subscribed to in your Feast Serving deployment? Can you show us your subscription configuration?

If you do see data in Redis but you are still not seeing results, then perhaps its related to your max_age. If your max_age is too small then older results will be ignored.

Reply:
Hi @Jeffwan sorry there is another part that needs to be updated in the sample Python code that publishes a FeatureRow to Kafka. Feature set reference in the FeatureRow should include the project prefix. Otherwise the ingestion will skip the message.

```diff
# docs/contributing.md#223
 
-# The format is [FEATURE_NAME]:[VERSION]
-row.feature_set = "driver:1"
+# The format is [PROJECT]/[FEATURE_NAME]:[VERSION]
+row.feature_set = "your_project_name/driver:1"
```

Reply:
@woop Yeah. I can see direct runner starts. I follow exact same subscription configuration in the master branch. 0.4.2-SNAPSHOT. Thanks for the debug guidance and it's pretty helpful. 

@davidheryanto  Change to "your_project_name/driver:1" works for me. Thanks!

```
2020-01-22 17:59:10.738 AUDIT 88e9fe523941.ant.amazon.com --- [pool-5-thread-1] f.c.l.AuditLogger                        : {action=SUBMIT, detail=Building graph and submitting to DirectRunner, id=kafka-to-serving1579744750734, resource=JOB, timestamp=Wed Jan 22 17:59:10 PST 2020}
2020-01-22 17:59:12.836  INFO 88e9fe523941.ant.amazon.com --- [pool-5-thread-1] f.i.ImportJob                            : Starting import job with settings:
Current Settings:
  appName: DirectRunnerJobManager
  blockOnRun: false
```

Now I am very confident this works for everyone trying master or 0.4.x branch. Thanks for the help!